app/src/behaviors/behavior_sticky_key.c

```diff
@@ -247,14 +265,33 @@ static int sticky_key_keycode_state_changed_listener(const zmk_event_t *eh) {
                 sticky_key->modified_key_usage_page == ev_copy.usage_page &&
                 sticky_key->modified_key_keycode == ev_copy.keycode) {
                 stop_timer(sticky_key);
-                release_sticky_key_behavior(sticky_key, ev_copy.timestamp);
+                sticky_keys_to_release_after_reraise[i] = sticky_key;
             }
         }
     }
-    if (event_reraised) {
-        return ZMK_EV_EVENT_CAPTURED;
+
+    // give each sticky key a chance to press their behavior before the event is reraised
+    for (int i = 0; i < ZMK_BHV_STICKY_KEY_MAX_HELD; i++) {
+        struct active_sticky_key *sticky_key = sticky_keys_to_press_before_reraise[i];
+        if (sticky_key) {
+            press_sticky_key_behavior(sticky_key, ev_copy.timestamp);
+        }
     }
-    return ZMK_EV_EVENT_BUBBLE;
+    // give each sticky key a chance to release their behavior after the event is reraised, lazily
+    // reraising. keep track whether the event has been reraised so we only reraise it once
+    bool event_reraised = false;
+    for (int i = 0; i < ZMK_BHV_STICKY_KEY_MAX_HELD; i++) {
+        struct active_sticky_key *sticky_key = sticky_keys_to_release_after_reraise[i];
+        if (sticky_key) {
+            if (!event_reraised) {
+                ZMK_EVENT_RAISE_AFTER(eh, behavior_sticky_key);
+                event_reraised = true;
+            }
+            release_sticky_key_behavior(sticky_key, ev_copy.timestamp);
+        }
```

Comment:
I favor early continue to avoid deeper nesting:

```suggestion
        if (!sticky_key) {
            continue;
        }
        
        if (!event_reraised) {
            ZMK_EVENT_RAISE_AFTER(eh, behavior_sticky_key);
            event_reraised = true;
        }
        release_sticky_key_behavior(sticky_key, ev_copy.timestamp);
```